### PR TITLE
Visualizer improvements pt1

### PR DESCRIPTION
**Pull request by djdameln**
# Description
First part of the visualizer refactor. Improves the looks of the images, especially for classification.
Example image for classification:
000_cls
Example image for segmentation:
000_seg

Also improves the logic within the visualizer callback class.

## Fixes
- #289 

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks! The output looks much better now.

File: anomalib/models/padim/config.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ dataset:
   format: mvtec
   path: ./datasets/MVTec
   category: bottle
-  task: segmentation
+  task: classification
```
Comment (samet-akcay) [inline anomalib/models/padim/config.yaml:6]: Why this change?

File: anomalib/utils/callbacks/visualizer_callback.py [hunk 2]
```diff
@@ -133,25 +139,26 @@ def on_test_batch_end(
...
-
-            image_classified = visualizer.add_text(
-                image=image,
-                text=f"""Pred: { "anomalous" if pred_score > threshold else "normal"}({pred_score:.3f}) \n
-                GT: {"anomalous" if bool(gt_label) else "normal"}""",
-            )
-            visualizer.add_image(image=image_classified, title="Classified Image")
+            visualizer = Visualizer()
+
+            if self.task == "segmentation":
```
[the remark this answers is not in the document]
Comment (djdameln) [reply anomalib/utils/callbacks/visualizer_callback.py:144]: It's because we only want to show masks when the task type is segmentation. Otherwise, when the task type is classification, we show the classification style visualization.

**Review by samet-akcay — APPROVED** [review]
